oci/spec_default_linux.go

```diff
@@ -0,0 +1,236 @@
+package oci
+
+import (
+	"os"
+	"strings"
+
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func iPtr(i int64) *int64        { return &i }
+func u32Ptr(i int64) *uint32     { u := uint32(i); return &u }
+func fmPtr(i int64) *os.FileMode { fm := os.FileMode(i); return &fm }
+
+func newDefaultCaps() []string {
+	return []string{
+		"CAP_CHOWN",
+		"CAP_DAC_OVERRIDE",
+		"CAP_FOWNER",
+		"CAP_FSETID",
+		"CAP_KILL",
+		"CAP_SETGID",
+		"CAP_SETUID",
+		"CAP_SETPCAP",
+		"CAP_AUDIT_WRITE",
+		"CAP_NET_BIND_SERVICE",
+		"CAP_NET_RAW",
+		"CAP_SYS_CHROOT",
+		"CAP_MKNOD",
+		"CAP_SETFCAP",
+		"CAP_LINUX_IMMUTABLE", //add
+		"CAP_NET_BROADCAST",   //add
+		"CAP_NET_ADMIN",       //add
+		"CAP_IPC_LOCK",        //add
+		"CAP_IPC_OWNER",       //add
+		"CAP_SYS_MODULE",      //add
+		"CAP_SYS_PTRACE",      //add
+		"CAP_SYS_PACCT",       //add
+		"CAP_SYS_ADMIN",       //add
+		"CAP_SYS_NICE",        //add
+		"CAP_SYS_RESOURCE",    //add
+		"CAP_SYS_TTY_CONFIG",  //add
+		"CAP_LEASE",           //add
+		"CAP_AUDIT_CONTROL",   //add
+		"CAP_SYSLOG",          //add
+		"CAP_WAKE_ALARM",      //add
+		"CAP_BLOCK_SUSPEND",   //add
+		"CAP_DAC_READ_SEARCH", //add
+	}
+}
+
+// NewDefaultSpec create a new spec.
+func NewDefaultSpec() *specs.Spec {
+	s := &specs.Spec{
+		Version: specs.Version,
+	}
+	s.Mounts = []specs.Mount{
+		{
+			Destination: "/proc",
+			Type:        "proc",
+			Source:      "proc",
+			Options:     []string{"nosuid", "noexec", "nodev"},
+		},
+		{
+			Destination: "/dev",
+			Type:        "tmpfs",
+			Source:      "tmpfs",
+			Options:     []string{"nosuid", "strictatime", "mode=755"},
+		},
+		{
+			Destination: "/dev/pts",
+			Type:        "devpts",
+			Source:      "devpts",
+			Options:     []string{"nosuid", "noexec", "newinstance", "ptmxmode=0666", "mode=0620", "gid=5"},
+		},
+		{
+			Destination: "/sys",
+			Type:        "sysfs",
+			Source:      "sysfs",
+			Options:     []string{"nosuid", "noexec", "nodev", "ro"},
+		},
+		{
+			Destination: "/dev/mqueue",
+			Type:        "mqueue",
+			Source:      "mqueue",
+			Options:     []string{"nosuid", "noexec", "nodev"},
+		},
+		// TODO(huamin): shm need to remove in default spec
+		{
+			Destination: "/dev/shm",
+			Type:        "tmpfs",
+			Source:      "shm",
+			Options:     []string{"nosuid", "noexec", "nodev", "mode=1777", "size=65536k"},
+		},
+	}
+
+	if _, err := os.Stat("/proc/self/ns/net"); err == nil {
+		s.Mounts = append(s.Mounts, specs.Mount{
+			Destination: "/sys/fs/cgroup",
+			Type:        "cgroup",
+			Source:      "cgroup",
+			Options:     []string{"ro", "nosuid", "noexec", "nodev"},
+		})
+	}
+
+	if defaultCaps := os.Getenv("default_caps"); defaultCaps != "" {
```

Comment:
who set the `default_caps` env ?

Reply:
Consider whether we need this, if not, I can remove this.